alibi/explainers/tests/test_shap_wrappers.py

```diff
@@ -1578,6 +1530,9 @@ def setter(obj: object, attr: str, val: Any):
         assert explanation.meta.keys() == DEFAULT_META_TREE_SHAP.keys()
         assert explanation.data.keys() == DEFAULT_DATA_TREE_SHAP.keys()
 
+        # parameter keys
+        assert set(explanation.meta['params']) <= set(TREE_SHAP_PARAMS)
```

Comment:
See comment in KernelShap tests.

Reply:
`TreeSHAP` does not set the following: `approximate`, `explain_loss`, `interactions`, `model_output`.

Reply:
I'm not sure why, the following lines of code set the first 3.

```python
self._update_metadata(
    {'interactions': self.interactions,
     'explain_loss': True if y is not None else False,
     'approximate': self.approximate,
},
     params=True,
)
```

Reply:
`model_output` seems to have ended up in `data` (just as `link`), so I'll move it to params in the patch. Not sure why we made a decision to put them in `data` originally (and didn't change `params`) but looking at it now, it seems like this fits better in `params`.

Reply:
I just realised that the `_update_metadata` is actually performed in `explain` which does not get called.